sys/include/net/gnrc/ipv6/nib/conf.h

```diff
@@ -134,7 +134,9 @@ extern "C" {
  */
 #ifndef CONFIG_GNRC_IPV6_NIB_ADV_ROUTER
 #if CONFIG_GNRC_IPV6_NIB_ROUTER && \
-    (!CONFIG_GNRC_IPV6_NIB_6LR || CONFIG_GNRC_IPV6_NIB_6LBR)
+    (!CONFIG_GNRC_IPV6_NIB_6LR || CONFIG_GNRC_IPV6_NIB_6LBR) && \
+    !(IS_USED(MODULE_GNRC_DHCPV6_CLIENT_IA_PD) || IS_USED(MODULE_GNRC_UHCPC) || \
+      IS_USED(MODULE_GNRC_IPV6_AUTO_SUBNETS))
 #define CONFIG_GNRC_IPV6_NIB_ADV_ROUTER               1
```

Comment:
Does it *ever* make sense to send router advertisements before having a prefix to advertise?

Reply:
Yes, if you do not have any prefixes in your network or don't want to use SLAAC, e.g.

Reply:
Also just supplanting the CFLAGS from the Makefile.dep to the correct place...

Reply:
Not objecting the PR, just wondering - because this regularly bites me when I flash `examples/gnrc_networking` and wonder why it takes so long to get a prefix, only to remember that I have to manually set `-rtr_adv` so the node sends router solicitations.

Reply:
> Not objecting the PR, just wondering - because this regularly bites me when I flash `examples/gnrc_networking` and wonder why it takes so long to get a prefix, only to remember that I have to manually set `-rtr_adv` so the node sends router solicitations.

You could also just use a non-routing host application, such as `examples/gcoap` or remove the `_router` in `gnrc_ipv6_router_default` in the application's Makefile ;-). The problem here isn't that the router tries to advertise, it is that you are using a router when you really want a non-routing host ;-).